### ISSUE TITLE
Error Console Displays "Could not find step name:XXX withdraw planes?"

### ISSUE THREAD

**Issue by ron-murhammer** [triplea-game/triplea#1954]
https://forums.triplea-game.org/topic/73/iron-war-official-thread/174

Problem from a code perspective is that this logic is run before casualties:
https://github.com/triplea-game/triplea/blob/566a37a767026363a9725be1325db5fbc1e4155b/src/main/java/games/strategy/triplea/delegate/MustFightBattle.java#L673

Then this logic is run after casualty selection and if there were non-amphib land units before casualties but aren't any after then it selects the "withdraw planes" retreat option that doesn't exist on the battle window display to highlight:
https://github.com/triplea-game/triplea/blob/566a37a767026363a9725be1325db5fbc1e4155b/src/main/java/games/strategy/triplea/delegate/MustFightBattle.java#L1443

**Comment by ron-murhammer** [comment]
There are 2 ways I can think of to fix this:
1. Make the retreat text in the battle window steps list generic for all types of retreats and just have the actual retreat pop up then specify the type/details.
2. Have the logic check if the necessary retreat type isn't available in the steps list and update the option. So in this case it would update "withdraw non-amphibious?" to "withdraw planes?" then highlight it.

I'm leaning towards option 1 as its simpler and I don't think listing the type of retreat in the steps list really gives much benefit especially since it can change as you take casualties.

**Comment by Cernelius** [comment]
By the way, an annoying issue, ever existing, that makes the "Partial Amphibious Retreat" non-serious and wrongly implemented is that TripleA obliges you to lose all unloaded units of the same kind before the not unloaded one, while, as far as I know, there should not be such a restriction.
I sometimes find myself in v3 doing stupid things like using artillery only from transports and taking armours as casualties before artillery just because, for example, to assure that some units will remain taking the territory after I retreat all the rest, and the engine doesn't allow me to manage that (usually when I attack Caucasus with Germans in v3, and want to take it, but only with some units, minimising the number that I will lose on the Russians counterattack thereafter).
This issue is particularly annoying in games having many different units, say Napoleonic Empires, in which this dynamic of distributing different units so to get the retreat as you wish turns really very silly.
This is the main reasons why, so far, I've decided to not use this property in a map I'm making, but if, say, as it should correctly work, I would be able to choose for each unit type if to lose an unloaded or a not unloaded one, then I'm thinking I would use this property, even to the fact that troops can't retreat to their own transports is so absurd that kind of makes me think to just impede retreat to all then... Duno really.

Just to give you a proof of this (that the fact that TripleA obliges you to lose unloaded units before not unloaded ones, in v3, is wrong), I just googled this. Of course, there can be better references, but I think this one should be enough.

http://www.axisandallies.org/forums/index.php?topic=12601.300

> _You can take the casualties on any units that you like.  You don't have to lose all of the amphibiously assaulting units before retreating the other ones.  All of the land-based units (and any air units) can retreat together at the end of any round of combat, leaving any remaining seaborne units behind to finish the fight alone.  This is the only time that a partial retreat is allowed.  However, if you do it, all of the land-based land units and air units must retreat together, leaving only seaborne units behind._

Or you can also take a look at, like, the official rulebook of WW2v5, if you know what I mean (just search for "retreat"), that will tell you that you have to keep overland and seaborne units separate on the battle strip, while this engine mixes them all and forcefully makes you take the seaborne ones first, amongst the ones of the same kind (thus the silliness that experienced players distribute the different types so to obtain an effect close to being actually able to take some not-unloaded units as casualties before all unloaded ones).

Basically, if the engine would work correctly here, it should present you with, like, separate overland and seaborne infantry etc., and allow you to take as casualties one or the other for each kind (and you should also always be able to see what is overland and what is seaborne, for the units you have left, at any point during combat).

So, what I'm saying, is that if you are going to rewrite some logic maybe also allow for taking overland before seaborne, if wanted, for the same unit type, and keep them separate in selection, as it should be.

**Comment by Cernelius** [comment]
But I'm not finding right now something that specifically confirms what I said, in that you should be able to select unloaded or not unloaded for a same type at will (while TripleA obliges you to lose the unloaded ones first). Maybe @panther2 or someone can double check and confirm. This has always been an annoying limit / bug, if I'm right that it is (maybe I'm missing some rules somewhere, but I don't think so).

**Comment by Cernelius** [comment]
Well, and there is also the issue that you may have an "isMarine" unit that it is stronger when unloaded, thus you would want to take the not unloaded marines before the unloaded ones, here, and you can't. Of course, this is mostly for custom maps, as you don't have marines in the regular v3 etc.. For example, the marines of Napoleonic Empires.

**Comment by ron-murhammer** [comment]
@Cernelius If you find that it is a bug then please open an issue with a way to reproduce it and reference to the rule. I'd like to keep this issue focused on just the display bug regarding various retreat types.
